regression/check_submodules.sh

```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
+
+filter='.[]'
+filter+='|select(.name == \"$path\")'
+filter+='|select(.name == \"$name\")'
```

Comment:
Why is it that we are checking that both the `$path` and the `$name` are equal to the wit submodule name? Git submodules may not always be installed at the root directory, so it's probably safer to check only the name?

Also, could you add a comment describing where these variables come from. I'm afraid that this is a pretty advanced usage of `git submodule foreach` and of `jq`, and to someone not experienced with the more advanced of both the tools, it's not exactly clear which of the two tools is setting the variables. (I eventually figured out that it's `git submodule foreach` that is defining them.)

Reply:
I added `$path` as a way of enforcing that submodules are at the root, but if we don't want to check that then I'll remove this

Reply:
also I don't think this script actually works since `.travis.yml` has `submodules: false` set so this would always pass. should I just parse the `.gitmodules` file instead?

Reply:
or just do a `git submodule update` since `regression/Makefile` will do it anyway?

Reply:
There must be some other script that takes care of checking out the submodules, because otherwise the other Travis CI tests couldn't possibly be working. You could try to invoke that script before running this one? I imagine we don't tell Travis CI to clone all submodules because there was a time when rocket-chip depended on `riscv-tools`, which depended on `riscv-gnu-toolchain`, which depended on the real GCC repo, which would have pulled down a ton of unnecessary code and git history.

It looks like `git submodule status` will still work even if you haven't initialized your submodules. It's not as nice as having the commit hash and the submodule name pulled into variables, but it shouldn't be too difficult to parse. Here's what it looks like to me:

```sh
$ git submodule status 
-e1aa5f3f5c0cdeb204047c3ca50801d9f7ea25f1 chisel3
-228c9a4b7432ac52178d63b8f27fe064aec71e9c firrtl
-70c1e1dd0954bb8e9d62563b8c9730bbc39d2bf0 hardfloat
-77195ab12aefc373ca688e0a9c4d710c13191341 torture
```

Reply:
Also, the `.gitmodules` file doesn't contain the actual commit hashes, only the (initial) remote locations of the submodules.